File: code_blocks/integrations/third-party-integrations/customerio_8.dart
import 'package:purchases_flutter/purchases_flutter.dart';
import 'package:customer_io/customer_io.dart';

Future<void> setupRevenueCatWithCustomerIo() async {
  // Configure RevenueCat first
  final configuration = PurchasesConfiguration('public_sdk_key')
    ..appUserId = 'your_app_user_id';
  await Purchases.configure(configuration);

  // Ensure Customer.io is initialized and identify has been called
  final customerIoId = CustomerIO.instance.profileIdentifier;

  if (customerIoId != null) {
    await Purchases.setAttributes({'\$customerioId': customerIoId});
  }
}

File: code_blocks/integrations/third-party-integrations/customerio_8.ts
import Purchases from '@revenuecat/purchases-react-native';
import CustomerIO from '@customerio/react-native';

export async function setupRevenueCatWithCustomerIo() {
  // Configure RevenueCat first
  await Purchases.configure({ apiKey: 'public_sdk_key', appUserID: 'your_app_user_id' });

  // Ensure Customer.io is initialized and identify has been called
  const customerIoId = await CustomerIO.getIdentifier();

  if (customerIoId) {
    await Purchases.setAttributes({ '$customerioId': customerIoId });
  }
}

